### ISSUE TITLE
Infinitely growing connection list in AmqpClientImpl

### ISSUE THREAD

**Issue by vladykin** [vert-x3/vertx-amqp-client#63]
AmqpConnections created with AmqpClientImpl.connect() are added to 
```java
private final List<AmqpConnection> connections = new CopyOnWriteArrayList<>();
```
but never removed, even when AmqpConnection is closed.

This leads to memory leak and eventually OOME in a scenario, when a single AmqpClient creates and immediately closes a new AmqpConnection every several seconds. Example use case - checking connectivity with message broker.

### Version

Version 3.9.6 that came as dependency of Quarkus 1.13.4.

**Comment by vietj** [comment]
fixed in master and backported to 3.9 https://github.com/vert-x3/vertx-amqp-client/pull/64

**Comment by vietj** [comment]
thanks for reporting
